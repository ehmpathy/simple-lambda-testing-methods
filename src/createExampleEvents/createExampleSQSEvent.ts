import { SQSEvent } from 'aws-lambda';

import { EVENT_METADATA_WHICH_SHOULD_NOT_BE_NEEDED } from './eventMetadataWhichShouldNotBeNeeded';

export const createExampleSQSEvent = ({
  messages,
}: {
  messages: string[];
}): SQSEvent => ({
  Records: messages.map((message) => {
    return {
      messageId: EVENT_METADATA_WHICH_SHOULD_NOT_BE_NEEDED,
      receiptHandle: EVENT_METADATA_WHICH_SHOULD_NOT_BE_NEEDED,
      body: message,
      attributes: EVENT_METADATA_WHICH_SHOULD_NOT_BE_NEEDED,
      messageAttributes: EVENT_METADATA_WHICH_SHOULD_NOT_BE_NEEDED,
      md5OfBody: EVENT_METADATA_WHICH_SHOULD_NOT_BE_NEEDED,
      eventSource: EVENT_METADATA_WHICH_SHOULD_NOT_BE_NEEDED,
      eventSourceARN: EVENT_METADATA_WHICH_SHOULD_NOT_BE_NEEDED,
      awsRegion: EVENT_METADATA_WHICH_SHOULD_NOT_BE_NEEDED,
    };
  }),
});
