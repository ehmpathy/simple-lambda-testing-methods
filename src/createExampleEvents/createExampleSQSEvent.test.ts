import { SQSEvent } from 'aws-lambda';
import { createExampleSQSEvent } from './createExampleSQSEvent';

describe('createExampleSQSEvent', () => {
  it('should create an accurate looking sns event', () => {
    const event: SQSEvent = createExampleSQSEvent({ messages: ['hello', 'there'] });
    expect(event).toMatchSnapshot();
  });
});
