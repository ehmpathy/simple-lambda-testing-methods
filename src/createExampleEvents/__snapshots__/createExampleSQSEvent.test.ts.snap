// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createExampleSQSEvent should create an accurate looking sns event 1`] = `
{
  "Records": [
    {
      "attributes": undefined,
      "awsRegion": undefined,
      "body": "hello",
      "eventSource": undefined,
      "eventSourceARN": undefined,
      "md5OfBody": undefined,
      "messageAttributes": undefined,
      "messageId": undefined,
      "receiptHandle": undefined,
    },
    {
      "attributes": undefined,
      "awsRegion": undefined,
      "body": "there",
      "eventSource": undefined,
      "eventSourceARN": undefined,
      "md5OfBody": undefined,
      "messageAttributes": undefined,
      "messageId": undefined,
      "receiptHandle": undefined,
    },
  ],
}
`;
